superset-frontend/src/features/databases/DatabaseModal/ExtraOptions.tsx

```diff
@@ -220,6 +220,22 @@ const ExtraOptions = ({
                 />
               </div>
             </StyledInputContainer>
+            <StyledInputContainer>
+              <div className="input-container">
+                <IndeterminateCheckbox
```

Comment:
Why are you using an indeterminate checkbox if you only have 2 possible states? Isn't the `Checkbox` component sufficient for your use case?

Reply:
Not sure, I just copied how the other checkboxes here were implemented; I'm not 100% clear on what the `indeterminate` functionality is for, though I notice they're all using `indetermine={false}` here as well.

Reply:
I notice `Checkbox` is missing the `title` and `labelText` properties that `IndeterminateCheckbox` has, so I suspect it's just being used for the label + tooltip here.

Reply:
Ah `Checkbox` also doesn't have an `id`, which the logic here needs to collect this data into a payload.

Reply:
Thank you for the additional context. It looks like we need to revisit these components given the inconsistencies. `IndeterminateCheckbox` shouldn't have a `indeterminate` property.